### DISCUSSION TITLE
useFormState improvement

### DISCUSSION THREAD

**Discussion by vlad-svr** [react-hook-form/react-hook-form#8690] [General]
### Version Number
7.33.1

**Is your feature request related to a problem? Please describe.**
I would like this hook to be able to return the state of the form by index, for example if we pass such a name
useFormState({
    control, name: "test.0" OR "test.0.lastName",
  });
  as the getFieldState function do it

**Describe the solution you'd like**
Now I'm going use getFieldState but would be great if useFormState hook has this able

**Reproduce**
1.  Add a couple of fields
2. Click submit form
3. See the console
Besides, I noticed that the lastname field(Controller) is missing in the form state

**Additional context**
Here is an example:
https://codesandbox.io/s/useformstate-issue-rhpxvm?file=/src/index.js
i am trying to get formState by name with index but i get all formState. We can see in console. Maybe it's a bug

<img width="1091" alt="image" src="https://user-images.githubusercontent.com/70812780/179052663-6d5e49cf-dbb9-4671-a9cc-f65396cbf37e.png">

### Thank you very much!

**Comment by bluebill1049** [comment]
could you save your codesandbox? it's empty will take a look at that first, in case it's a bug.

**Comment by vlad-svr** [comment]
oh sorry I didn't save :( I'll create it again soon

**Comment by vlad-svr** [comment]
https://codesandbox.io/s/useformstate-issue-rhpxvm?file=/src/index.js
to reproduce
1. add a couple of fields
2. click submit form
3. See the console

Besides, I noticed that the lastname field(Controller) is missing in the form state

**Comment by bluebill1049** [comment]
you never register an input with `lastName` that's why.

**Comment by bluebill1049** [comment]
I am going to convert this into a discussion question, we can discuss there if there is any need to introduce a new feature 👍
